File: 2022/go/1.go
package main

import (
	"fmt"
	"os"
	"bufio"
	"strconv"
	"sort"
)

func main() {

	input, err := os.Open("../1.txt")
	if err != nil {
		fmt.Printf("Could not read the file due to this %s error \n", err)
	}
	sc := bufio.NewScanner(input)

	var totals []int
	var cur_index int = 0
	var cur_value int = 0
	for sc.Scan() {
		if v := len(sc.Text()); v > 0 {
			val, _ := strconv.Atoi(sc.Text())
			cur_value += val
		} else {
            totals = append(totals, cur_value)
            cur_index += 1
            cur_value = 0
		}
	}
	sort.Ints(totals)
	fmt.Printf("Max Value: %v\n", totals[len(totals) - 1])	
	top3 := totals[len(totals) - 1] + totals[len(totals) - 2] + totals[len(totals) - 3]
	fmt.Printf("Top 3 Value: %v\n", top3)	

}

File: 2022/rust/1.rs
use std::fs;


fn main() {
    let input = fs::read_to_string("../1.txt").unwrap();
    let lines : Vec<&str> = input.split("\n").collect();

    let mut cur_value : i32 = 0;
    let mut totals : Vec<i32> = Vec::new();
    let mut cur_index : i32 = 0;
    for line in lines {
        if (line.len() > 0) {
            cur_value += line.parse::<i32>().unwrap();
        }
        else {
            totals.push(cur_value);
            cur_index += 1;
            cur_value = 0;
        }
    }
    totals.sort();
    println!("Max Value: {}",  totals[totals.len() - 1]);
    let top3 = totals[totals.len() - 1] + totals[totals.len() - 2] + totals[totals.len() - 3];
    println!("Top 3 Value: {}",  top3);
}